src/Pages/SingleVideo/index.js

```diff
@@ -0,0 +1,203 @@
+import { useParams, Link } from "react-router-dom";
+import { useState } from "react";
+import { useAuthData, useVideo, useWatchLater, useLike } from "../../Context";
+import YouTube from "react-youtube";
+import { PlaylistChoose } from "../VideoList/Components/PlaylistChoose";
+import { addLikes, removeFromLikes } from "../../ApiService/Liked";
+import { addWatchlater, removeWatchlater } from "../../ApiService";
+
+import { ToastContainer, toast } from "react-toastify";
+
+export const SingleVideo = () => {
+  const [selectPlaylist, setSelectPlaylist] = useState(false);
+  const {
+    userAuth: { isUserLoggedIn },
+  } = useAuthData();
+  const { VideoDispatch } = useVideo();
+  const { watchLaterDispatch } = useWatchLater();
+  const { LikeDispatch } = useLike();
+
+  const { videoid } = useParams();
+  const { VideoState } = useVideo();
+  const { data: ListOFVideo } = VideoState;
+  const chooseVideo = ListOFVideo.find((item) => item._id === videoid);
+
+  const RelatedVideo = ListOFVideo.slice(0, 5);
+
+  const opts = {
+    height: "600",
+    width: "800",
+    playerVars: {
+      autoplay: 0,
+    },
+  };
+
+  const addWatchLaterHandler = async (videoDetails) => {
+    // add to watchlater using context
+    const token = localStorage.getItem("encodedToken");
+    //call api service
+    const { data, status } = await addWatchlater(videoDetails, token);
+    if (status === 200 || status === 201) {
+      watchLaterDispatch({
+        type: "SET_WISHLIST",
+        payload: data.watchlater,
+      });
+      toast.success("Add video to the Watch Later list", {
+        position: "bottom-right",
+        autoClose: 3000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+    if (data === undefined) {
+      const {
+        status: removeStatuse,
+        data: { watchlater },
+      } = await removeWatchlater(videoDetails._id, token);
+      if (removeStatuse === 200 || removeStatuse === 201) {
+        watchLaterDispatch({
+          type: "SET_WISHLIST",
+          payload: watchlater,
+        });
+      }
+
+      toast.warning("Remove video from the Watch Later List", {
+        position: "bottom-right",
+        autoClose: 3000,
+        hideProgressBar: false,
+        closeOnClick: true,
+        pauseOnHover: true,
+        draggable: true,
+        progress: undefined,
+      });
+    }
+
```

Comment:
here toast properties you giving , directly can give in ToastContainer Component , so you don't need to write it repeatedly,